src/ergtable.h

```diff
@@ -0,0 +1,140 @@
+#ifndef ERGTABLE_H
+#define ERGTABLE_H
+
+#include <QList>
+#include <QSettings>
+#include <QObject>
+#include <cmath>
+
+#include <QMetaType>
+#include <QDebug>
+
+class DataPoint {
+public:
+    double cadence = 0; // RPM, default initialized
+    double wattage = 0; // Watts, default initialized
+    double resistance = 0; // Some unit, default initialized
+
+    DataPoint() = default; // Default constructor
+
+    DataPoint(double c, double w, double r) : cadence(c), wattage(w), resistance(r) {} // Custom constructor
+};
+
+Q_DECLARE_METATYPE(DataPoint)
+
+class ergTable : public QObject {
+    Q_OBJECT
+
+public:
+    ergTable(QObject *parent = nullptr) : QObject(parent) {
+        loadSettings();
+    }
```

Comment:
I'm wondering why you've made this a QObject, with all that comes with it, but never use said functionality. Are you planning to have signals and slots?

trxappgateusbbike::_ergtable is never passed a parent in its construction.
If you put this code in this constructor and run the tests, it never throws because parent is always null.
```
        if(parent!=nullptr) {
            throw "Got a parent";
        }
```

Reply:
yes i made this for future implementation. now it's not necessary of course :)